source/MRViewer/MRSurfaceManipulationWidget.cpp

```diff
@@ -604,4 +628,71 @@ void SurfaceManipulationWidget::updateRegionUVs_( const VertBitSet& region )
     obj_->setAncillaryUVCoords( std::move( uvs ) );
 }
 
+void SurfaceManipulationWidget::updateValueChanges_( const VertBitSet& region )
+{
+    const auto& oldPoints = lastStableObjMesh_->mesh()->points;
+    const auto& points = obj_->mesh()->points;
+    const auto& mesh = *obj_->mesh();
+    BitSetParallelFor( region, [&] ( VertId v )
+    {
+        const Vector3f shift = points[v] - oldPoints[v];
+        const float sign = dot( shift, mesh.normal( v ) ) >= 0.f ? 1.f : -1.f;
+        valueChanges_[v] = lastStableValueChanges_[v] + shift.length() * sign;
+    } );
+
+    const auto& [min, max] = parallelMinMax( valueChanges_.vec_ );
+    changesMinVal_ = min;
+    changesMaxVal_ = max;
+
+    updateRegionUVs_( region );
+}
+
+void SurfaceManipulationWidget::updateValueChangesByDistance_( const VertBitSet& region )
+{
+    const auto& mesh = obj_->mesh();
+    const auto& meshVerts = mesh->points;
+
+    std::vector<MeshProjectionResult> projResults( meshVerts.size() );
+    BitSetParallelFor( region, [&] ( VertId v )
+    {
+        projResults[v] = findProjection( meshVerts[v], *originalMesh_, FLT_MAX, nullptr, 0 );
```

Comment:
use shift.lengthSq instead of FLT_MAX ?

Reply:
this only works if we compare unmodified and shifted meshes, but this method compares the final and original meshes